app/vmalert/datasource/vm_graphite_api.go

```diff
@@ -64,12 +64,14 @@ func (s *VMStorage) setGraphiteReqParams(r *http.Request, query string, timestam
 	}
 	q.Set("format", "json")
 	q.Set("target", query)
-	from := "-5min"
-	if s.lookBack > 0 {
-		lookBack := timestamp.Add(-s.lookBack)
-		from = strconv.FormatInt(lookBack.Unix(), 10)
+	if !q.Has("from") {
```

Comment:
I'd suggest changing the order of applying expected params and extra params. Please see how it is done in prometheus implementation.

Reply:
Updated flow to make it similar to Prometheus one:
1) Sets all default values
2) Merges values with provided `extraParams`